### DISCUSSION TITLE
Install from source not working with custom formula?

### DISCUSSION THREAD

**Discussion by tmewett** [Homebrew/discussions#147] [Everyday usage] [answered]
Hi, within the last 30 days my GitHub Actions job involving the pre-installed Homebrew stopped working. I was wondering if someone could have a look and check I'm not aware of some recent change.

I am trying to run this build step:

    brew install --build-from-source macos/sdl2.rb
    brew install sdl2_image dylibbundler

You can see the full workflow file here: [build.yml @ 60a49c](https://github.com/tmewett/BrogueCE/blob/60a49cdb9b2e37aef81531c651e4a06fa6449460/.github/workflows/build.yml). And my modified formula is here: [sdl2.rb @ 60a49c](https://github.com/tmewett/BrogueCE/blob/60a49cdb9b2e37aef81531c651e4a06fa6449460/macos/sdl2.rb). I just added a few lines to the start of *install* method. It's based on the latest core formula at the time of posting.

What now happens is, the first command seemingly does nothing. The first output appearing in my logs tells me that the SDL2 bottle is being downloaded... explicitly what I told brew to not do. This makes me think the command is silently doing nothing, and the SDL2 bottle is being installed to satisfy SDL2_image.

      brew install --build-from-source macos/sdl2.rb
      brew install sdl2_image dylibbundler
      shell: /bin/bash -e {0}
    ==> Downloading https://homebrew.bintray.com/bottles/sdl2-2.0.12_1.catalina.bottle.1.tar.gz
    ...
    ==> Downloading https://homebrew.bintray.com/bottles/sdl2_image-2.0.5.catalina.bottle.tar.gz
    ...

[Link to the full job logs](https://github.com/tmewett/BrogueCE/runs/1408459080?check_suite_focus=true#step:3:1).

In fact, these changes follow my first run from today, where instead of using `--build-from-source` I had a custom formula with a deleted `bottle do ... end` section. So I'm not sure it's an issue with that option.

Additionally, I tested just the from-source command on its own. [There is no output at all](https://github.com/tmewett/BrogueCE/runs/1408871468?check_suite_focus=true#step:3:2). And I know it's not silently completing, since otherwise the bottle wouldn't be downloaded in the above job.

Grateful for any ideas. If it's a bug, I can open an issue. Thanks!

**Comment by dtrodrigues** [comment, marked as answer]
I think it has to do with the logic of detecting if the argument to install is a formula name, cask, path, tap etc. Can you try doing `brew install --build-from-source ./macos/sdl2.rb` as a workaround?

The way you're currently doing it looks like it worked through brew 2.5.2, tries to tap `/usr/local/Homebrew/Library/Taps/macos/homebrew-sdl2.rb` in 2.5.3, and now does nothing in 2.5.4 and later.

**Reply by tmewett** [reply]
You're right - [adding `./` fixed it](https://github.com/tmewett/BrogueCE/runs/1411555028?check_suite_focus=true#step:3:1). I also found Homebrew/brew#8966 based on your suggestion. Unfortunately it seems to be closed as wontfix, despite the lack of any output or failure. I hope at least this aspect gets fixed in a future version. Thanks for your help
